src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/BulkUpdateStudyInstance.sql

```diff
@@ -0,0 +1,157 @@
+
+    WHILE (@rowsUpdated < @totalCount)
+    BEGIN
+        
+        DELETE FROM @updatedInstances
+
+        UPDATE TOP (@batchSize) dbo.Instance
+        SET LastStatusUpdatedDate = @currentDate, OriginalWatermark = Watermark, Watermark = NewWatermark, NewWatermark = NULL
+        OUTPUT deleted.PartitionKey, @studyInstanceUid, deleted.SeriesInstanceUid, deleted.SopInstanceUid, deleted.StudyKey, deleted.SeriesKey, deleted.InstanceKey, deleted.NewWatermark  INTO @updatedInstances
+        WHERE PartitionKey = @partitionKey
+            AND StudyInstanceUid = @studyInstanceUid
+            AND Status = 1
+            AND NewWatermark IS NOT NULL
+
+        SET @rowsUpdated = @rowsUpdated + @@ROWCOUNT;
+
+        UPDATE EQT
+        SET Watermark = U.Watermark
+        FROM ExtendedQueryTagString EQT
+        JOIN @updatedInstances U 
+        ON EQT.SopInstanceKey1 = U.StudyKey
+        AND EQT.SopInstanceKey2 = U.SeriesKey
+        AND EQT.SopInstanceKey3 = U.InstanceKey
+        AND EQT.PartitionKey = @partitionKey
+        AND EQT.ResourceType = @imageResourceType
+
+        UPDATE EQT
+        SET Watermark = U.Watermark
+        FROM ExtendedQueryTagLong EQT
+        JOIN @updatedInstances U 
+        ON EQT.SopInstanceKey1 = U.StudyKey
+        AND EQT.SopInstanceKey2 = U.SeriesKey
+        AND EQT.SopInstanceKey3 = U.InstanceKey
+        AND EQT.PartitionKey = @partitionKey
+        AND EQT.ResourceType = @imageResourceType
+
+        UPDATE EQT
+        SET Watermark = U.Watermark
+        FROM ExtendedQueryTagDouble EQT
+        JOIN @updatedInstances U 
+        ON EQT.SopInstanceKey1 = U.StudyKey
+        AND EQT.SopInstanceKey2 = U.SeriesKey
+        AND EQT.SopInstanceKey3 = U.InstanceKey
+        AND EQT.PartitionKey = @partitionKey
+        AND EQT.ResourceType = @imageResourceType
+
+        UPDATE EQT
+        SET Watermark = U.Watermark
+        FROM ExtendedQueryTagDateTime EQT
+        JOIN @updatedInstances U 
+        ON EQT.SopInstanceKey1 = U.StudyKey
+        AND EQT.SopInstanceKey2 = U.SeriesKey
+        AND EQT.SopInstanceKey3 = U.InstanceKey
+        AND EQT.PartitionKey = @partitionKey
+        AND EQT.ResourceType = @imageResourceType
+
+        UPDATE EQT
+        SET Watermark = U.Watermark
+        FROM ExtendedQueryTagPersonName EQT
+        JOIN @updatedInstances U 
+        ON EQT.SopInstanceKey1 = U.StudyKey
+        AND EQT.SopInstanceKey2 = U.SeriesKey
+        AND EQT.SopInstanceKey3 = U.InstanceKey
+        AND EQT.PartitionKey = @partitionKey
+        AND EQT.ResourceType = @imageResourceType
+
+        -- Insert into change feed table for update action type
+        INSERT INTO dbo.ChangeFeed
+        (TimeStamp, Action, PartitionKey, StudyInstanceUid, SeriesInstanceUid, SopInstanceUid, OriginalWatermark, CurrentWatermark)
+        SELECT @currentDate, 2, PartitionKey, StudyInstanceUid, SeriesInstanceUid, SopInstanceUid, Watermark, Watermark
```

Comment:
Should'nt this be OriginalWatermrk?

Reply:
For changefeed both original and currentwatermark are equal means it is created. We are creating a new row so I kept the same.

Reply:
Updated